packages/library-base/src/commonMain/kotlin/io/realm/internal/InternalMutableRealm.kt

```diff
@@ -37,7 +36,7 @@ internal interface InternalMutableRealm : MutableRealm {
                     "they can be queried this way. Use `MutableRealm.copyToRealm()` to turn it into " +
                     "a managed object."
             )
-        } else if (!obj.isFrozen()) {
+        } else if ((obj as RealmObjectInternal).`$realm$Owner` == realmReference) {
```

Comment:
Isn't this a different check than `isFrozen`? But not sure I fully grasp the implications.

Reply:
It isn't sufficient to check if it is live because it could theoretically be from another live realm or a completely different realm.